tests/components/mqtt/test_sensor.py

```diff
@@ -311,7 +311,7 @@ async def test_setting_sensor_value_via_mqtt_json_message(
     async_fire_mqtt_message(hass, "test-topic", '{ "val": "" }')
     state = hass.states.get("sensor.test")
 
-    assert state.state == ""
+    assert state.state == STATE_UNKNOWN
```

Comment:
Once I thought the same about handling `""` as `None` but I had to revert that as it gave issues:
See: https://github.com/home-assistant/core/pull/81347

Reply:
My adjustment only applies when a valid number is expected (ie. it has a state class, or a uom, or a number device class).

Blank string will still be allowed for other sensors.

Reply:
I would expect the state to be "" here, this is not a number.

Reply:
The UOM is the reason why a number is expected.
I'll remove the UOM instead...